server/fcm/connector.go

```diff
@@ -198,22 +199,51 @@ func (conn *Connector) GetPrefix() string {
 
 // ServeHTTP handles the subscription in FCM
 func (conn *Connector) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if r.Method != http.MethodPost && r.Method != http.MethodDelete {
-		logger.WithField("method", r.Method).Error("Only HTTP POST and DELETE methods supported.")
+	if r.Method != http.MethodPost && r.Method != http.MethodDelete && r.Method != http.MethodGet {
+		logger.WithField("method", r.Method).Error("Only HTTP POST, GET and DELETE methods supported.")
 		http.Error(w, `{"error":"method not allowed"}`, http.StatusMethodNotAllowed)
 		return
 	}
-	userID, fcmID, topic, err := conn.parseParams(r.URL.Path)
+	userID, fcmID, unparsedPath, err := conn.parseUserIdAndDeviceId(r.URL.Path)
 	if err != nil {
 		http.Error(w, `{"error":"invalid parameters in request"}`, http.StatusBadRequest)
 		return
 	}
 	switch r.Method {
 	case http.MethodPost:
+		topic, err := conn.parseTopic(unparsedPath)
+		if err != nil {
+			http.Error(w, `{"error":"invalid parameters in request"}`, http.StatusBadRequest)
+			return
+		}
 		conn.addSubscription(w, topic, userID, fcmID)
 	case http.MethodDelete:
+		topic, err := conn.parseTopic(unparsedPath)
+		if err != nil {
+			http.Error(w, `{"error":"invalid parameters in request"}`, http.StatusBadRequest)
+			return
+		}
 		conn.deleteSubscription(w, topic, userID, fcmID)
+	case http.MethodGet:
+		conn.retriveAllSubscription(w, userID, fcmID)
+	}
+}
+
+func (conn *Connector) retriveAllSubscription(w http.ResponseWriter, userID, fcmID string) {
+	topics := make([]string, 0)
+
+	for k, v := range conn.subscriptions {
+		logger.WithField("key", k).Info("retriveAllSubscription")
+		if v.route.Get(applicationIDKey) == fcmID && v.route.Get(userIDKey) == userID {
+			logger.WithField("path", v.route.Path).Info("retriveAllSubscription path")
+			topics = append(topics, string(v.route.Path))
+		}
+	}
+	serialized, err := json.Marshal(topics)
```

Comment:
If interested you can use `json.Encoder` to directly write in a Writer the encoding.
See the `Encoder` doc in `encoding/json` package: https://godoc.org/encoding/json